scripts/core/editor3/reducers/suggestions.jsx

```diff
@@ -436,7 +437,16 @@ const setDeleteSuggestionForCharacter = (editorState, data) => {
     if (currentData != null && currentData.type === 'ADD_SUGGESTION' &&
         currentData.author === data.author) {
         // if current character already a suggestion of current user, delete the character
-        return deleteCharacter(editorState);
+        let newState = deleteCharacter(editorState);
+
+        // also delete the suggestion if it was the last character of that suggestion
```

Comment:
It will be better if you move this code on deleteCharacter, because this code should run every time deleteCharacter is called.